### PR TITLE
feature(persistence): Implemented DynamoStore.get_filtered_documents() method

### LINKED ISSUE #17: def get_filtered_documents(self, symbol_to_find: str = None, target_date: datetime.date = None) and tests
def get_filtered_documents(self, symbol_to_find: str = None, target_date: datetime.date = None):
        """
        Returns a list of documents matching given ticker and/or date
        :param symbol_to_find: ticker as a string
        :param target_date: desired date as a datetime.date, leave empty to get for all available dates
        :return: a list of dicts() each containing data available for a stock for a given period of time
        """
   Keep in mind that scan will be slow - you have to rely on indexes and search operations to make your implementation fast.

### PR DESCRIPTION
**Pull request by astepanovv**
- Implemented function get_filtered_documents() in DynamoStore class.

Closes #17

### CODE REVIEW

File: persistence/DynamoStore.py [hunk 1]
```diff
@@ -15,14 +19,67 @@ def store_documents(self, documents: list):
...
+
+            while True:
+                last_evaluated_key = response.get("LastEvaluatedKey")
+                if last_evaluated_key:
+                    if condition_expression:
+                        response = self.Table.query(
+                            KeyConditionExpression=condition_expression,
+                            ExclusiveStartKey=last_evaluated_key
+                        )
+                    else:
```
Comment (petelind) [inline persistence/DynamoStore.py:68]: @astepanovv , not sure I'm getting in which cases we may need scan() here. Missing index = cannot search => try scanning instead?..

**Comment by petelind** [comment]
@astepanovv , Please also resolve conflicts before pulling, ok?
Put your implementation over one we have now, marking it as get_filtered_documents_2()
Overall - excellent, concise and easy to understand.

File: persistence/DynamoStore.py (hunk 1 above)
Comment (astepanovv) [reply persistence/DynamoStore.py]: fixed

File: persistence/DynamoStore.py [hunk 2]
```diff
@@ -50,14 +52,80 @@ def get_filtered_documents(self, symbol_to_find: str = None, target_date: dateti
...
+                    output.Results = "Target date must be datetime.date!"
+                    return output
+
+                date_expression = Key("date").eq(str(target_date))
+                condition_expression = date_expression if not condition_expression \
+                    else condition_expression & date_expression
+
+            query_params: dict = {
+                "TableName": self.TableName,
+                "KeyConditionExpression": condition_expression
```
Comment (AndreiStepanov1TR) [inline persistence/DynamoStore.py:98]: Add KeyConditionalExpression only if condition_expression not None

File: persistence/DynamoStore.py [hunk 3]
```diff
@@ -50,14 +52,80 @@ def get_filtered_documents(self, symbol_to_find: str = None, target_date: dateti
...
+                if items:
+                    total_items.extend(items)
+
+                last_evaluated_key = result.get("LastEvaluatedKey")
+                if not last_evaluated_key:
+                    break
+
+            output = Results()
+            output.ActionStatus = ActionStatus.SUCCESS
+            output.Results = items
```
Comment (AndreiStepanov1TR) [inline persistence/DynamoStore.py:121]: output.Results = total_items

File: persistence/DynamoStore.py [hunk 4]
```diff
@@ -50,14 +52,80 @@ def get_filtered_documents(self, symbol_to_find: str = None, target_date: dateti
...
-            output.Results = criteria.query(self.table)
+            output.Results = criteria.query(self.Table)
             output.ActionStatus = ActionStatus.SUCCESS
             return output
         except Exception as e:
             catastrophic_exception = AppException(ex=e, message='Catastrophic failure when trying to query symbols '
                                                                 'for the Dynamo!')
             raise catastrophic_exception
 
+    def get_filtered_documents_v2(self, symbol_to_find: str = None, target_date: date = None) -> Results:
```
Comment (AndreiStepanov1TR) [inline persistence/DynamoStore.py:63]: Need unit tests

**Comment by petelind** [comment]
The majority voted to keep the OO-style implementation, sorry.

**Pull request closed without merging**